### PR TITLE
feat: [PL-28099]: added new API to clear git sync sdk cache

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes

Added new API to clear git-sync cache across Pipeline service and NG manager service (for connectors). This is an API that will only be used for internal purposes. It is majorly required for migration of git-sync projects to new gitX.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37346)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37346)

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java [hunk 1]
```diff
@@ -49,4 +63,27 @@ public GitEnabledDTO getGitEnabledDTO(
     }
     return gitEnabledDTOBuilder.build();
   }
+
+  public void resetGitSyncSDKCache(String accountIdentifier, String orgIdentifier, String projectIdentifier) {
+    List<String> errors = new ArrayList<>();
+    List<EntityType> supportedEntityTypes = Arrays.asList(CONNECTORS, PIPELINES);
```
Comment (bhavya181) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java:69]: templates is also a supported entity right?

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java` line 69 at r1](https://reviewable.io/reviews/harness/harness-core/37346#-NBX01Z7dvhMk9KDx-x3-r1-69:-NBXA5wT8bR9R0EvnWyU:b-unv0lv) ([raw file](https://github.com/harness/harness-core/blob/0b406d31e175b90049911268844106b4cd4e9a7e/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java#L69)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

templates is also a supported entity right?
</blockquote></details>

yeah, but to start with, we only want to support migration for pipeline and input-sets and connectors. Templates will come for migration may be in a month.

**Comment by mohitg0795** [comment]
trigger messagecheck

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 7 of 9 files at r1, 1 of 2 files at r2, 7 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37346#-:-NBppEaM7Wu_ArA8pxjl:bnfp4nl)* status: 15 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> Reviewed 7 of 9 files at r1, 1 of 2 files at r2, 7 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37346)* status: 1 change requests. 1 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java` line 69 at r1](https://reviewable.io/reviews/harness/harness-core/37346#-NBX01Z7dvhMk9KDx-x3-r1-69:-NBqGpBn28A-nQGAkBHs:bfkr1kk) ([raw file](https://github.com/harness/harness-core/blob/0b406d31e175b90049911268844106b4cd4e9a7e/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java#L69)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

yeah, but to start with, we only want to support migration for pipeline and input-sets and connectors. Templates will come for migration may be in a month.
</blockquote></details>

Then how will those templates work?

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java` line 69 at r1](https://reviewable.io/reviews/harness/harness-core/37346#-NBX01Z7dvhMk9KDx-x3-r1-69:-NBqXfUR3eivhqky_MGP:b7zqvju) ([raw file](https://github.com/harness/harness-core/blob/0b406d31e175b90049911268844106b4cd4e9a7e/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java#L69)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Then how will those templates work?
</blockquote></details>

Thats out of scope for now. There will be no templates in the projects we cover for now.

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java` line 69 at r1](https://reviewable.io/reviews/harness/harness-core/37346#-NBX01Z7dvhMk9KDx-x3-r1-69:-NBqe9ABDxqBtZsFfh_U:b-edzej0) ([raw file](https://github.com/harness/harness-core/blob/0b406d31e175b90049911268844106b4cd4e9a7e/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java#L69)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Thats out of scope for now. There will be no templates in the projects we cover for now.
</blockquote></details>

This API is only for migration for now. So, we don't want to include things that we are not covering in scope of migration. Once we plan to migrate templates as well, it will be added into the API.

**Review by bhavya181 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37346)* status: :shipit: complete! ✓

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java` line 69 at r1](https://reviewable.io/reviews/harness/harness-core/37346#-NBX01Z7dvhMk9KDx-x3-r1-69:-NBqzSEh3qe7zJ_PRhyU:bgk1ngi) ([raw file](https://github.com/harness/harness-core/blob/0b406d31e175b90049911268844106b4cd4e9a7e/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java#L69)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

This API is only for migration for now. So, we don't want to include things that we are not covering in scope of migration. Once we plan to migrate templates as well, it will be added into the API.
</blockquote></details>

ok, so, we are migrating only those projects, in which templates are not present right?

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java` line 69 at r1](https://reviewable.io/reviews/harness/harness-core/37346#-NBX01Z7dvhMk9KDx-x3-r1-69:-NBsQpcO4ZhetMy-CbvM:b2nhswy) ([raw file](https://github.com/harness/harness-core/blob/0b406d31e175b90049911268844106b4cd4e9a7e/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/GitEnabledHelper.java#L69)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

ok, so, we are migrating only those projects, in which templates are not present right?
</blockquote></details>

Yes

**Pull request merged**